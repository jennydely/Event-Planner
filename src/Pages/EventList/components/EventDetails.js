import React, { useContext } from 'react'
import { animated } from 'react-spring'
import styled from 'styled-components/macro'
import Checkbox from '../../components/common/Checkbox'
import Paragraph from '../../components/common/Paragraph'
import trashIcon from '../../../img/trashIcon.svg'
import eyeIcon from '../../../img/eyeIcon.svg'
import hideEyeIcon from '../../../img/hideEyeIcon.svg'
import routeIcon from '../../../img/routeIcon.svg'
import websiteIcon from '../../../img/websiteIcon.svg'
import gCalendarIcon from '../../../img/gCalendarIcon.svg'
import getColorOfEventCategory from '../../../services/getColorOfEventCategory'
import { UserContext } from "../../../providers/UserProvider";

export default function EventDetails({
  event,
  style,
  bind,
  id,
  onHideButtonClick,
  onDeleteButtonClick,
  onTicketCheckboxClick,
}) {
  const { poster, name, street, zip, location, price, website, eventStartDate, eventEndDate } = event
  const defaultImg =
    'https://delyed.de/wp-content/uploads/2018/01/5d737e918441914a9d2743268ef65439.jpg'
  const user = useContext(UserContext);

  return (
    <Details name={event.category} style={style} {...bind}>
      <LinkPoster href={poster ? poster : website} target="blank">
        {' '}
        <EventPoster
          src={poster ? poster : defaultImg}
          alt={name + ' Poster'}
        />
      </LinkPoster>
      <Address>Address: </Address>
      <Price>Price: </Price>
      <Name>{name}</Name>
      <ParagraphColumn3 row={2}>{street}</ParagraphColumn3>
      <ParagraphColumn3 row={3}>
        {zip} {location.length > 15 ? location.slice(0, 15) + '...' : location}
      </ParagraphColumn3>
      <ParagraphColumn3 row={5}>
        {price ? price + ' €' : 'kostenlos'}{' '}
      </ParagraphColumn3>
      {user ?
        <><TicketLabel id="Ticket" price={price}>
          Ticket
      </TicketLabel>

          <Ticket
            type="checkbox"
            checked={event.ticketBought}
            onChange={handleCheckboxClick}
            htmlFor="Ticket"
            price={price}
          />
        </> : ''}

      <ButtonContainer row={7} space={'space-around'}>
        <a href={website} target="blank" title="link">
          <img src={websiteIcon} alt="website" />
        </a>
        <a
          href={`https://www.google.de/maps/dir//+${encodeURIComponent(
            street
          )},+${encodeURIComponent(zip)},${encodeURIComponent(location)}`}
          target="blank"
          title="link"
        >
          <img src={routeIcon} alt="route" />
        </a>
        <a href={`http://www.google.com/calendar/render?
action=TEMPLATE
&text=${name}
&dates=${eventStartDate.split('-').join('')}/${eventEndDate.split('-').join('')}
&details=${encodeURIComponent(price + ' € ' + website)}
&location=${encodeURIComponent(street + ',' + zip + ',' + location)}
&trp=false
&sprop=
&sprop=name:`}
          target="_blank" rel="nofollow">
          <img src={gCalendarIcon} alt="google calendar" /></a>

        {user ?
          <><HideButton onClick={handleHideButtonClick} id={id}>
            {event.isHidden ? (
              <img src={eyeIcon} alt="show" />
            ) : (
                <img src={hideEyeIcon} alt="hide" />
              )}
          </HideButton>
            <DeleteButton onClick={handleDeleteButtonClick}>
              <img src={trashIcon} alt="delete" />
            </DeleteButton></>
          : ''}
      </ButtonContainer>
    </Details>
  )

  function handleDeleteButtonClick() {
    if (window.confirm('Are you sure you wish to delete this item?'))
      onDeleteButtonClick(event.id)
  }

  function handleCheckboxClick() {
    onTicketCheckboxClick(event.id)
  }
  function handleHideButtonClick() {
    onHideButtonClick(event.id)
  }
}

const Details = styled(animated.section)`
  display: grid;
  grid-template-columns: 30% repeat(2, auto) 30%;
  grid-template-rows: repeat(8, auto);
  margin: 0;
  margin-top: -2px;
  border-radius: 7px;
  background: var(--eventDetails);
  border: 2px solid ${(opt) => getColorOfEventCategory(opt.name)};
  border-top: 0;
  overflow: hidden;

  &::before,
  &::after {
    content: '';
    position: absolute;
    left: 0;
    padding: 2px;
    width: 100%;
    height: 100%;
  }

  &::before {
    height: 0;
  }

  &::after {
    height: 8px;
    bottom: 0;
  }
`
const ParagraphColumn3 = styled(Paragraph)`
  grid-column: 3/5;
  grid-row: ${(props) => props.row};
  text-align: left;
`
const LinkPoster = styled.a`
  grid-column: 1;
  grid-row: 1 / span 5;
  padding: 0;
  margin: 11px 4px 0 4px;
`
const EventPoster = styled.img`
  grid-column: 1 / span 1;
  grid-row: 1 / span 5;
  margin: 0;
  padding: 0;
  object-fit: cover;
  width: 90%;
  max-height: 90%;
`
const Address = styled(Paragraph)`
  grid-column: 2;
  grid-row: 1 / span 4;
  margin-top: 7px;
  margin-left: 4px;
  font-weight: 400;
  font-size: 112.5%;
`
const Price = styled(Paragraph)`
  grid-column: 2;
  grid-row: 5;
  font-weight: 400;
  font-size: 112.5%;
`
const TicketLabel = styled.label`
  grid-column: 2;
  grid-row: 6;
  text-align: right;
  align-self: center;
  margin: 2px;
  font-weight: bold;
  font-size: 112.5%;
  display: ${({ price }) => (price ? '' : 'none')};
`
const Name = styled(Paragraph)`
  grid-column: 3/5;
  grid-row: 1;
  text-align: left;
  margin-top: 7px;
`
const Ticket = styled(Checkbox)`
  grid-column: 3;
  grid-row: 6;
  display: ${({ price }) => (price ? '' : 'none')};
`
const ButtonContainer = styled.div`
  grid-column: 1/5;
  grid-row: ${(props) => props.row};
  display: flex;
  width: 100%;
  margin: 1px;
  padding: 0;
  padding-top: 7px;
  justify-content: ${(props) => props.space};
`
const HideButton = styled.button`
  display: inline-block;
  margin: 0;
`
const DeleteButton = styled.button`
  display: inline-block;
  margin: 0;
`
