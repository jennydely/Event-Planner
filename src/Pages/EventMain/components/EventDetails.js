import React from 'react'
import { animated } from 'react-spring'
import styled from 'styled-components/macro'
import { useHistory } from 'react-router-dom'
import Checkbox from '../../components/common/Checkbox'
import Paragraph from '../../components/common/Paragraph'
import trashIcon from '../../../img/trashIcon.svg'
import packlistIcon from '../../../img/packlistIcon.svg'
import routeIcon from '../../../img/routeIcon.svg'
import websiteIcon from '../../../img/websiteIcon.svg'
import getColorOfEventCategory from '../../../services/getColorOfEventCategory'

export default function EventDetails({
  event,
  style,
  bind,
  onDeleteButtonClick,
  onTicketCheckboxClick,
}) {
  const {
    poster,
    name,
    street,
    zip,
    location,
    price,
    website,
    packlistCategory,
  } = event
  const defaultImg =
    'https://delyed.de/wp-content/uploads/2018/01/5d737e918441914a9d2743268ef65439.jpg'
  const history = useHistory()
  function showPacklist() {
    history.push('/packlist/' + packlistCategory)
  }

  return (
    <Details name={event.category} style={style} {...bind}>
      <LinkPoster href={poster ? poster : website} target="blank">
        {' '}
        <EventPoster
          src={poster ? poster : defaultImg}
          alt={name + ' Poster'}
        />
      </LinkPoster>
      <Address>Address: </Address>
      <Price>Price: </Price>
      <Name>{name}</Name>
      <ParagraphColumn3 row={2}>{street}</ParagraphColumn3>
      <ParagraphColumn3 row={3}>
        {zip} {location}
      </ParagraphColumn3>
      <ParagraphColumn3 row={5}>
        {price ? price + ' â‚¬' : 'kostenlos'}{' '}
      </ParagraphColumn3>
      <TicketLabel id="Ticket" price={price}>
        Ticket
      </TicketLabel>
      <Ticket
        type="checkbox"
        checked={event.ticketBought}
        onChange={handleCheckboxClick}
        htmlFor="Ticket"
        price={price}
      />
      <ButtonContainer row={7} space={'space-around'}>
        <PacklistButton onClick={showPacklist}>
          <img src={packlistIcon} alt="packlist" />
        </PacklistButton>
        <a href={website} target="blank" title="link">
          <img src={websiteIcon} alt="website" />
        </a>
        <a
          href={`https://www.google.de/maps/dir//+${encodeURIComponent(
            street
          )},+${encodeURIComponent(zip)},${encodeURIComponent(location)}`}
          target="blank"
          title="link"
        >
          <img src={routeIcon} alt="route" />
        </a>
        <DeleteButton onClick={handleDeleteButtonClick}>
          <img src={trashIcon} alt="delete" />
        </DeleteButton>
      </ButtonContainer>
    </Details>
  )

  function handleDeleteButtonClick() {
    if (window.confirm('Are you sure you wish to delete this item?'))
      onDeleteButtonClick(event.id)
  }

  function handleCheckboxClick() {
    onTicketCheckboxClick(event.id)
  }
}

const Details = styled(animated.section)`
  display: grid;
  grid-template-columns: 20% repeat(4, auto) 20%;
  grid-template-rows: repeat(8, auto);
  margin: 0;
  margin-top: -2px;
  border: 2px solid ${(opt) => getColorOfEventCategory(opt.name)};
  border-top: 0;
  overflow: hidden;

  &::before,
  &::after {
    content: '';
    position: absolute;
    left: 0;
    width: 100%;
    height: 100%;
  }

  &::before {
    height: 0;
  }

  &::after {
    height: 8px;
    bottom: 0;
  }
`
const ParagraphColumn3 = styled(Paragraph)`
  grid-column: 3/7;
  grid-row: ${(props) => props.row};
  text-align: left;
`
const LinkPoster = styled.a`
  grid-column: 1;
  grid-row: 1 / span 5;
`
const EventPoster = styled.img`
  grid-column: 1 / span 1;
  grid-row: 1 / span 5;
  margin: 7px;
  object-fit: cover;
  width: 90%;
  max-height: 90%;
`
const Address = styled(Paragraph)`
  grid-column: 2;
  grid-row: 1 / span 4;
  margin-top: 7px;
  margin-left: 4px;
  font-weight: 400;
  font-size: 112.5%;
`
const Price = styled(Paragraph)`
  grid-column: 2;
  grid-row: 5;
  font-weight: 400;
  font-size: 112.5%;
`
const TicketLabel = styled.label`
  grid-column: 2;
  grid-row: 6;
  text-align: right;
  align-self: center;
  margin: 2px;
  font-weight: bold;
  font-size: 112.5%;
  display: ${({ price }) => (price ? '' : 'none')};
`
const Name = styled(Paragraph)`
  grid-column: 3/7;
  grid-row: 1;
  text-align: left;
  margin-top: 7px;
`
const Ticket = styled(Checkbox)`
  grid-column: 3;
  grid-row: 6;
  display: ${({ price }) => (price ? '' : 'none')};
`
const ButtonContainer = styled.div`
  grid-column: 1/7;
  grid-row: ${(props) => props.row};
  display: flex;
  width: 100%;
  margin: 1px;
  padding: 0;
  padding-top: 7px;
  justify-content: ${(props) => props.space};
`
const PacklistButton = styled.button`
  display: inline-block;
  margin: 0;
`
const DeleteButton = styled.button`
  display: inline-block;
  margin: 0;
`
