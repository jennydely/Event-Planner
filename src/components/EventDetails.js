import React from 'react'
import { animated } from 'react-spring'
import styled from 'styled-components/macro'
import { useHistory } from 'react-router-dom'
import Checkbox from '../common/Checkbox'
import Paragraph from '../common/Paragraph'
import getColorByName from '../services/getColorByName'
import trashIcon from '../img/trash.svg'

export default function EventDetails({
  event,
  style,
  bind,
  onDeleteButtonClick,
}) {
  const {
    poster,
    name,
    street,
    zip,
    location,
    price,
    website,
    packlistCategory,
  } = event
  const defaultImg =
    'https://delyed.de/wp-content/uploads/2018/01/5d737e918441914a9d2743268ef65439.jpg'
  const history = useHistory()
  function handleBackButtonClick() {
    history.push('/packlist/' + packlistCategory)
  }

  return (
    <Details name={event.category} style={style} {...bind}>
      <LinkPoster href={poster ? poster : website} target="blank">
        {' '}
        <EventPoster
          src={poster ? poster : defaultImg}
          alt={name + ' Poster'}
        />
      </LinkPoster>
      <Address>Address: </Address>
      <Price>Price: </Price>
      <Name>{name}</Name>
      <ParagraphColumn3 row={2}>{street}</ParagraphColumn3>
      <ParagraphColumn3 row={3}>
        {zip} {location}
      </ParagraphColumn3>
      <ParagraphColumn3 row={5}>
        {price ? price + ' â‚¬' : 'kostenlos'}{' '}
      </ParagraphColumn3>
      <TicketLabel id="Ticket" price={price}>
        Ticket
      </TicketLabel>
      <Ticket type="checkbox" htmlFor="Ticket" price={price} />
      <ButtonContainer>
        <button onClick={handleBackButtonClick}>PackList</button>
        <ExternalLink href={website} target="blank" title="link">
          Website
        </ExternalLink>
        <ExternalLink
          href="https://www.google.de/maps"
          target="blank"
          title="link"
        >
          Route
        </ExternalLink>
      </ButtonContainer>
      <DeleteButton onClick={handleDeleteButtonClick}>
        <img src={trashIcon} alt="delete" />
      </DeleteButton>
    </Details>
  )
  function handleDeleteButtonClick() {
    onDeleteButtonClick(event.id)
  }
}

const Details = styled(animated.section)`
  display: grid;
  grid-template-columns: 20% repeat(4, auto);
  grid-template-rows: repeat(8, auto);
  margin: 0;
  margin-top: -2px;
  border: 2px solid ${(opt) => getColorByName(opt.name)};
  border-top: 0;
  overflow: hidden;

  &::before,
  &::after {
    content: '';
    position: absolute;
    left: 0;
    width: 100%;
    height: 100%;
  }

  &::before {
    height: 0;
  }

  &::after {
    height: 8px;
    bottom: 0;
  }
`
const ParagraphColumn3 = styled(Paragraph)`
  grid-column: 3/6;
  grid-row: ${(props) => props.row};
  text-align: left;
`
const LinkPoster = styled.a`
  grid-column: 1;
  grid-row: 1 / span 5;
`
const EventPoster = styled.img`
  grid-column: 1 / span 1;
  grid-row: 1 / span 5;
  margin: 7px;
  object-fit: cover;
  width: 90%;
  max-height: 90%;
`
const Address = styled(Paragraph)`
  grid-column: 2;
  grid-row: 1 / span 4;
  margin-top: 7px;
  font-weight: 400;
  font-size: 112.5%;
`
const Price = styled(Paragraph)`
  grid-column: 2;
  grid-row: 5;
  font-weight: 400;
  font-size: 112.5%;
`
const TicketLabel = styled.label`
  grid-column: 2;
  grid-row: 6;
  text-align: right;
  align-self: center;
  margin: 2px;
  font-weight: bold;
  font-size: 112.5%;
  display: ${({ price }) => (price ? '' : 'none')};
`
const Name = styled(Paragraph)`
  grid-column: 3/6;
  grid-row: 1;
  text-align: left;
  margin-top: 7px;
`
const Ticket = styled(Checkbox)`
  grid-column: 3;
  grid-row: 6;
  display: ${({ price }) => (price ? '' : 'none')};
`
const ButtonContainer = styled.div`
  grid-column: 1/6;
  grid-row: 7;
  display: flex;
  width: 100%;
  margin: 1px;
  padding: 0;
  justify-content: space-around;
`
const ExternalLink = styled.a`
  display: inline-block;
  min-width: fit-content;
  min-height: fit-content;
  margin: 2px;
  padding: 4px 6px;
  border: none;
  border-radius: 6px;
  background-color: var(--darkgrey-75);
  font-size: 130%;
  text-align: center;
  color: var(--sandyellow-main);
`
const DeleteButton = styled.button`
  grid-column: 5;
  grid-row: 8;
  color: var(--red-main);
  font-size: 130%;
  float: right;
  text-decoration: none;
  border-radius: 20px;
  background: none;
  margin: 4px;
  margin-top: 7px;
`
