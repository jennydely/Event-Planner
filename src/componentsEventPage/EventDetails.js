import React from 'react';
import { animated } from 'react-spring'
import styled from 'styled-components/macro'

export default function EventDetails({ event, style, bind }) {
  const { poster, name, street, zip, location, price, website } = event
  const defaultImg = "https://raw.githubusercontent.com/jennydely/Event-Planner/EventDetails/src/img/defaultImg.jpg"

  return (
    <Details name={event.category} style={style} {...bind}>
     <LinkPoster href={poster ? poster : website} target="blank"> <EventPoster src={poster ? poster : defaultImg} alt={name + ' Poster'} /></LinkPoster>
      <Address>Address: </Address>
      <Price>Price: </Price>
      <Name>{name}</Name>
      <Street>{street}</Street>
      <Location>{zip} {location}</Location>
      <PriceValue>{price} â‚¬</PriceValue>
      <ButtonContainer>
        <button onClick="/packList">PackList</button>
        <ExternalLink href={website} target="blank">Website</ExternalLink>
        <ExternalLink href="https://www.google.de/maps" target="blank">Googlemaps</ExternalLink>
      </ButtonContainer>
    </Details>
  )
}

const Details = styled(animated.section)`
display:grid;
grid-template-columns: 20% repeat(4,auto);
grid-template-rows: repeat(7,auto) 30px 30px;
margin:0;
margin-top: -2px;
border: 2px solid ${({name}) => name === 'sand' ? 'rgba(248,149,17,0.46)' : (name === 'metal' ? 'rgba(49,42,42,0.75)' : (name === 'medieval' ? 'rgba(67,40,24,0.70)' : (name === 'other' ? 'rgba(153,88,42,0.70)' : 'rgb(96,99,104)')))};
border-top: 0;
overflow: hidden;

&::before,
&::after {
  content: '';
  position: absolute;
  left: 0;
  width: 100%;
}

&::before {
  height: 0;
  }

&::after {
  height: 8px;
  bottom: 0;
}
`
const LinkPoster = styled.a`
grid-column: 1;
grid-row: 1 / span 5;
`
const EventPoster = styled.img`
grid-column: 1 /span 1;
grid-row: 1 / span 5;
margin:7px;
object-fit: cover;
width: 90%;
max-height: 90%;
`

const Address = styled.p`
grid-column: 2;
grid-row: 1/ span 4;
text-align:right;
margin:4px;
margin-top: 7px;
font-weight:bold;
font-size: 100%;
`
const Price = styled.p`
grid-column: 2;
grid-row: 5;
text-align:right;
margin:4px;
font-weight:bold;
font-size: 100%;
`
const Name = styled.p`
grid-column: 3/5;
grid-row: 1;
text-align:left;
margin:4px;
margin-top:7px;
font-size: 100%;
`
const Street = styled.p`
grid-column: 3/5;
grid-row: 2;
text-align:left;
margin:4px;
font-size: 100%;
`
const Location = styled.p`
grid-column: 3/5;
grid-row: 3;
text-align:left;
margin:4px;
font-size: 100%;
`
const PriceValue = styled.p`
grid-column: 3/5;
grid-row: 5;
text-align:left;
margin:4px;
font-size: 100%;
`
const ButtonContainer = styled.div`
grid-column: 1/6;
grid-row: 6;
display:flex;
with: 100%;
padding:3,5px;
justify-content:space-around;
`
const ExternalLink = styled.a`
display: inline-block; 
with:fit-content;
height:fit-content;
padding: 4px 7px;
border: solid 2px rgb(49,42,42);
border-radius: 6px;
background-color:rgba(49,42,42,0.75);
color: rgb(187,148,87);
font-size: 100%; 
text-align: center; 
`
