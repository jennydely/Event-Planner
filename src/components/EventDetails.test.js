import React from 'react'
import { render} from '@testing-library/react'
import renderer from 'react-test-renderer'
import EventDetails from './EventDetails'
import 'jest-styled-components'

describe('EventDetails', () => {
    const event = {
        "id": "7",
        "poster": "https://www.wacken-winter-nights.com/fileadmin/_processed_/csm_wwn_20_191024-web_6495c0af7f.jpg",
        "name": "WackenWinterNights",
        "street": "Reselithweg 17",
        "zip": "25596",
        "location": "Wacken",
        "price": "99",
        "ticketBought": false,
        "website": "https://www.wacken-winter-nights.com/",
        "eventStartDate": "2021-02-14",
        "eventEndDate": "2021-02-16",
        "category": "metal"
    }

    it('displays the eventdetails, like name, location & price', () => {
        const { getByText, getByRole, getAllByTitle } = render(<EventDetails event={event} />)
            
        expect(getByText(event.name)).toBeInTheDocument()
        expect(getByText(event.zip + ' ' + event.location)).toBeInTheDocument()
        expect(getByText(event.street)).toBeInTheDocument()
        expect(getByText(event.price + ' â‚¬')).toBeInTheDocument()
        //expect(getByRole('button')).toBeInTheDocument()
        expect(getAllByTitle('link')).toHaveLength(2)
        expect(getByRole('checkbox', { checked: false })).toBeInTheDocument()

    })
    it('renders correctly', () => {
        const tree = renderer.create(<EventDetails event={event} />)
        expect(tree).toMatchSnapshot()
    })
})