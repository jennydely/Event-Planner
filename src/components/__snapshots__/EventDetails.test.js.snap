// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetails renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c11 {
  min-width: 44px;
  min-height: 44px;
  text-align: right;
  margin: 2px;
  font-weight: bold;
  font-size: 100%;
}

.c3 {
  text-align: right;
  margin: 2px;
  font-size: 100%;
}

.c0 {
  display: grid;
  grid-template-columns: 20% repeat(4,auto);
  grid-template-rows: repeat(8,auto) 30px auto;
  margin: 0;
  margin-top: -2px;
  border: 2px solid var(--darkgrey-75);
  border-top: 0;
  overflow: hidden;
}

.c0::before,
.c0::after {
  content: '';
  position: absolute;
  left: 0;
  width: 100%;
  height: 100%;
}

.c0::before {
  height: 0;
}

.c0::after {
  height: 8px;
  bottom: 0;
}

.c7 {
  grid-column: 3/5;
  grid-row: 2;
  text-align: left;
}

.c8 {
  grid-column: 3/5;
  grid-row: 3;
  text-align: left;
}

.c9 {
  grid-column: 3/5;
  grid-row: 5;
  text-align: left;
}

.c1 {
  grid-column: 1;
  grid-row: 1 / span 5;
}

.c2 {
  grid-column: 1 / span 1;
  grid-row: 1 / span 5;
  margin: 7px;
  object-fit: cover;
  width: 90%;
  max-height: 90%;
}

.c4 {
  grid-column: 2;
  grid-row: 1 / span 4;
  margin-top: 7px;
  font-weight: 400;
  font-size: 112.5%;
}

.c5 {
  grid-column: 2;
  grid-row: 5;
  font-weight: 400;
  font-size: 112.5%;
}

.c10 {
  grid-column: 2;
  grid-row: 6;
  text-align: right;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 2px;
  font-weight: bold;
  font-size: 112.5%;
}

.c6 {
  grid-column: 3/5;
  grid-row: 1;
  text-align: left;
  margin-top: 7px;
}

.c12 {
  grid-column: 3;
  grid-row: 6;
}

.c13 {
  grid-column: 1/6;
  grid-row: 7;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin: 1px;
  padding: 0;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c14 {
  display: inline-block;
  min-width: -webkit-fit-content;
  min-width: -moz-fit-content;
  min-width: fit-content;
  min-height: -webkit-fit-content;
  min-height: -moz-fit-content;
  min-height: fit-content;
  margin: 2px;
  padding: 4px 6px;
  border: none;
  border-radius: 6px;
  background-color: var(--darkgrey-75);
  font-size: 130%;
  text-align: center;
  color: var(--sandyellow-main);
}

<body>
    <div>
      <section
        class="c0"
        name="metal"
      >
        <a
          class="c1"
          href="https://www.wacken-winter-nights.com/fileadmin/_processed_/csm_wwn_20_191024-web_6495c0af7f.jpg"
          target="blank"
        >
           
          <img
            alt="WackenWinterNights Poster"
            class="c2"
            src="https://www.wacken-winter-nights.com/fileadmin/_processed_/csm_wwn_20_191024-web_6495c0af7f.jpg"
          />
        </a>
        <p
          class="c3 c4"
        >
          Address: 
        </p>
        <p
          class="c3 c5"
        >
          Price: 
        </p>
        <p
          class="c3 c6"
        >
          WackenWinterNights
        </p>
        <p
          class="c3 c7"
        >
          Reselithweg 17
        </p>
        <p
          class="c3 c8"
        >
          25596
           
          Wacken
        </p>
        <p
          class="c3 c9"
        >
          99 €
           
        </p>
        <label
          class="c10"
          id="Ticket"
        >
          Ticket
        </label>
        <input
          class="c11 c12"
          for="Ticket"
          type="checkbox"
        />
        <div
          class="c13"
        >
          <button>
            PackList
          </button>
          <a
            class="c14"
            href="https://www.wacken-winter-nights.com/"
            target="blank"
            title="link"
          >
            Website
          </a>
          <a
            class="c14"
            href="https://www.google.de/maps"
            target="blank"
            title="link"
          >
            Route
          </a>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="EventDetails__Details-sc-5njpgn-0 bvWrsj"
      name="metal"
    >
      <a
        class="EventDetails__LinkPoster-sc-5njpgn-2 bTgyqo"
        href="https://www.wacken-winter-nights.com/fileadmin/_processed_/csm_wwn_20_191024-web_6495c0af7f.jpg"
        target="blank"
      >
         
        <img
          alt="WackenWinterNights Poster"
          class="EventDetails__EventPoster-sc-5njpgn-3 cnaSWE"
          src="https://www.wacken-winter-nights.com/fileadmin/_processed_/csm_wwn_20_191024-web_6495c0af7f.jpg"
        />
      </a>
      <p
        class="Paragraph-sc-1mi161p-0 EventDetails__Address-sc-5njpgn-4 byOeOf jeXSoW"
      >
        Address: 
      </p>
      <p
        class="Paragraph-sc-1mi161p-0 EventDetails__Price-sc-5njpgn-5 byOeOf fZtPPd"
      >
        Price: 
      </p>
      <p
        class="Paragraph-sc-1mi161p-0 EventDetails__Name-sc-5njpgn-7 byOeOf bDjoam"
      >
        WackenWinterNights
      </p>
      <p
        class="Paragraph-sc-1mi161p-0 EventDetails__ParagraphColumn3-sc-5njpgn-1 byOeOf hXNTKG"
      >
        Reselithweg 17
      </p>
      <p
        class="Paragraph-sc-1mi161p-0 EventDetails__ParagraphColumn3-sc-5njpgn-1 byOeOf kmtXNf"
      >
        25596
         
        Wacken
      </p>
      <p
        class="Paragraph-sc-1mi161p-0 EventDetails__ParagraphColumn3-sc-5njpgn-1 byOeOf gFLMdl"
      >
        99 €
         
      </p>
      <label
        class="EventDetails__TicketLabel-sc-5njpgn-6 dSspxY"
        id="Ticket"
      >
        Ticket
      </label>
      <input
        class="Checkbox-sc-1a8a6ug-0 EventDetails__Ticket-sc-5njpgn-8 KEjef fiAzfq"
        for="Ticket"
        type="checkbox"
      />
      <div
        class="EventDetails__ButtonContainer-sc-5njpgn-9 gSLsaQ"
      >
        <button>
          PackList
        </button>
        <a
          class="EventDetails__ExternalLink-sc-5njpgn-10 juBvfX"
          href="https://www.wacken-winter-nights.com/"
          target="blank"
          title="link"
        >
          Website
        </a>
        <a
          class="EventDetails__ExternalLink-sc-5njpgn-10 juBvfX"
          href="https://www.google.de/maps"
          target="blank"
          title="link"
        >
          Route
        </a>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
